app/javascript/components/workflow-credential-mapping-form/index.jsx

```diff
@@ -0,0 +1,131 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+
+import MiqFormRenderer from '@@ddf';
+import { CredentialMapperComponent } from './helper';
+import componentMapper from '../../forms/mappers/componentMapper';
+import createSchema from './workflow-credential-mapping-form.schema';
+import miqRedirectBack from '../../helpers/miq-redirect-back';
+
+const WorkflowCredentialMappingForm = ({ recordId }) => {
+  const [{
+    credentials,
+    credentialReferences,
+    payloadCredentials,
+    workflowAuthentications,
+    initialValues,
+    isLoading,
+  }, setState] = useState({ isLoading: !!recordId });
+  const submitLabel = !!recordId ? __('Save') : __('Add');
+
+  // custom component mapper
+  const mapper = {
+    ...componentMapper,
+    'credential-mapper': CredentialMapperComponent,
+  };
+
+  useEffect(() => {
+    // eslint-disable-next-line camelcase
+    API.get(
+      `/api/authentications?expand=resources&filter[]=type='ManageIQ::Providers::Workflows::AutomationManager::WorkflowCredential'`
+    ).then(({ resources }) => {
+      API.get(`/api/configuration_script_payloads/${recordId}`).then(({ name, payload, credentials }) => {
+        const initialCredentials = credentials != null ? credentials.credentials : {};
+
+        /*
+          Creates the list of credential references from the workflow payload by parcing each state and saving them to payloadCredentials.
+          Duplicate references get overridden.
+        */
+        const jsonPayload = JSON.parse(payload).States;
+        let payloadCredentials = {};
+
+        Object.keys(jsonPayload).forEach((key1) => {
+          if (jsonPayload[key1].Credentials != null) {
+            Object.keys(jsonPayload[key1].Credentials).forEach((key2) => {
+              payloadCredentials = {
+                [key2]: jsonPayload[key1].Credentials[key2],
+                ...payloadCredentials,
+              };
+            });
+          }
+        });
+
+        // Returns the user to the show_list page if the workflow has no credential references in it's payload
+        if (Object.keys(payloadCredentials).length === 0) {
+          throw __('Workflow does not have any credentials to map.');
```

Comment:
Can you show a screenshot of this?  Will this just dump a flash message to the screen?

Reply:
> Can you show a screenshot of this? Will this just dump a flash message to the screen?

yes

https://github.com/ManageIQ/manageiq-ui-classic/assets/64800041/e1b149c8-1832-4ec3-b949-e6a673faaab2

Reply:
Not necessarily for this PR but I wonder if we should disable the button if there are no credentials to map

Reply:
I agree with @agrare - perhaps even do both.

Reply:
talked with @agrare and we'll do this in a follow up pr